src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModelUpdater.java

```diff
@@ -229,7 +229,7 @@ public void start() {
    * Interrupts the updater and starts it again. This can be used to trigger an instant main view
    * model update, skipping the sleep interval.
    */
-  public void restart() {
+  public synchronized void restart() {
```

Comment:
Now when I think of it, it actually looks a bit suspicious that `start` ain't synchronized because it accesses a field that is set in a synced method `restart`. However, as the idea is that `restart` can only be called after `start` has been called ('after' in the sense of Java's _happens-before_ relation).

But is it so? I guess `start` is called almost immediately after the object is created but is there still a small time window where an external `restart` call could happen? What I suggest (not as a part of this PR, though - it is fine) is that we'd get rid of `start`/`restart` duality and only have one synced method with the following code:
```java
 Optional.ofNullable(thread).ifPresent(Thread::interrupt);
thread = new Thread(this::run);
thread.start();
```
And `thread` would be just `null` initially.

Reply:
That small time frame doesn't exist, because `start` is called in the `computeIfAbsent` function in `PluginSettings`, so the reference to the updater becomes "publicly available" only after `start` has already been called.

That being said, the current interface does allow that risk if the code using it isn't careful. I really like your suggested solution, and I think it's the way to go :+1: